### PR TITLE
Adding API1600 support for Networkset and Uplinkset

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Adding API1600 support for Networkset and Uplinkset

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/shared_samples/ethernet_network.rb [hunk 1]
```diff
@@ -32,6 +40,15 @@
 # api_version = 500 & variant = Synergy to OneviewSDK::API500::Synergy::EthernetNetwork
 # api_version = 600 & variant = C7000 to OneviewSDK::API600::C7000::EthernetNetwork
 # api_version = 600 & variant = Synergy to OneviewSDK::API600::Synergy::EthernetNetwork
+# api_version = 800 & variant = Synergy to OneviewSDK::API800::Synergy::EthernetNetwork
+# api_version = 800 & variant = C7000 to OneviewSDK::API800::C7000::EthernetNetwork
+# api_version = 1000 & variant = C7000 to OneviewSDK::API1000::C7000::EthernetNetwork
+# api_version = 1000 & variant = Synergy to OneviewSDK::API1000::Synergy::EthernetNetwork
+# api_version = 1200 & variant = C7000 to OneviewSDK::API1200::C7000::EthernetNetwork
+# api_version = 1200 & variant = C7000 to OneviewSDK::API1200::C7000::EthernetNetwork
```
Comment (chebroluharika) [inline examples/shared_samples/ethernet_network.rb:48]: This is repeated. I think it should be for Synergy Variant

File: examples/shared_samples/fc_network.rb [hunk 2]
```diff
@@ -32,6 +40,15 @@
 # api_version = 500 & variant = Synergy to OneviewSDK::API500::C7000::FCNetwork
 # api_version = 600 & variant = C7000 to OneviewSDK::API600::C7000::FCNetwork
 # api_version = 600 & variant = Synergy to OneviewSDK::API600::C7000::FCNetwork
+# api_version = 800 & variant = C7000 to OneviewSDK::API800::C7000::FCNetwork
+# api_version = 800 & variant = Synergy to OneviewSDK::API800::C7000::FCNetwork
```
Comment (chebroluharika) [inline examples/shared_samples/fc_network.rb:44]: It should be OneviewSDK::API800::Synergy::FCNetwork

File: examples/shared_samples/fc_network.rb [hunk 3]
```diff
@@ -32,6 +40,15 @@
...
+# api_version = 1000 & variant = C7000 to OneviewSDK::API1000::C7000::FCNetwork
+# api_version = 1000 & variant = Synergy to OneviewSDK::API1000::C7000::FCNetwork
```
Comment (chebroluharika) [inline examples/shared_samples/fc_network.rb:46]: It should be OneviewSDK::API1000::Synergy::FCNetwork

File: examples/shared_samples/fc_network.rb [hunk 4]
```diff
@@ -32,6 +40,15 @@
...
+# api_version = 1200 & variant = C7000 to OneviewSDK::API1200::C7000::FCNetwork
+# api_version = 1200 & variant = Synergy to OneviewSDK::API1200::C7000::FCNetwork
```
Comment (chebroluharika) [inline examples/shared_samples/fc_network.rb:48]: It should be OneviewSDK::API1200::Synergy::FCNetwork

File: examples/shared_samples/fc_network.rb [hunk 5]
```diff
@@ -32,6 +40,15 @@
...
+# api_version = 1600 & variant = C7000 to OneviewSDK::API1600::C7000::FCNetwork
+# api_version = 1600 & variant = Synergy to OneviewSDK::API1600::C7000::FCNetwork
```
Comment (chebroluharika) [inline examples/shared_samples/fc_network.rb:50]: It should be OneviewSDK::API1600::Synergy::FCNetwork

File: examples/shared_samples/network_set.rb [hunk 6]
```diff
@@ -32,7 +40,16 @@
...
 # api_version = 600 & variant = C7000 to OneviewSDK::API600::C7000::NetworkSet
 # api_version = 600 & variant = Synergy to OneviewSDK::API600::Synergy::NetworkSet
-
+# api_version = 800 & variant = C7000 to OneviewSDK::API800::C7000::NetworkSet
+# api_version = 800 & variant = Synergy to OneviewSDK::API800::Synergy::NetworkSet
+# api_version = 1000 & variant = C7000 to OneviewSDK::API1000::C7000::NetworkSet
+# api_version = 1000 & variant = Synergy to OneviewSDK::API1000::Synergy::NetworkSet
+# api_version = 1200 & variant = C7000 to OneviewSDK::API1200::C7000::NetworkSet
+# api_version = 1200 & variant = Synergy to OneviewSDK::API1200::Synergy::NetworkSet
+# api_version = 1600 & variant = C7000 to OneviewSDK::API1200::C7000::NetworkSet
```
Comment (AsisBagga) [inline examples/shared_samples/network_set.rb:49]: @yuvirani need to correct the Inheritance, should be OneviewSDK::API1600

File: examples/shared_samples/network_set.rb [hunk 7]
```diff
@@ -32,7 +40,16 @@
...
+# api_version = 1600 & variant = Synergy to OneviewSDK::API1200::Synergy::NetworkSet
```
Comment (AsisBagga) [inline examples/shared_samples/network_set.rb:50]: @yuvirani Same as above

File: examples/shared_samples/network_set.rb (hunk 6 above)
Comment (chebroluharika) [inline examples/shared_samples/network_set.rb:49]: It should be OneviewSDK::API1600::C7000::NetworkSet

File: examples/shared_samples/network_set.rb (hunk 7 above)
Comment (chebroluharika) [inline examples/shared_samples/network_set.rb:50]: It should be OneviewSDK::API1600::Synergy::NetworkSet

File: spec/unit/resource/api1600/synergy/uplink_set_spec.rb [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+RSpec.describe OneviewSDK::API1600::Synergy::UplinkSet do
+  include_context 'shared context'
+
+  it 'inherits from OneviewSDK::API1200::Synergy::UplinkSet' do
+    expect(described_class).to be < OneviewSDK::API1200::Synergy::UplinkSet
+  end
+
+  describe '#initialize' do
+    it 'should be initialize the instance with default values' do
+      item = described_class.new(@client_1000)
```
Comment (chebroluharika) [inline spec/unit/resource/api1600/synergy/uplink_set_spec.rb:12]: I think it should be @client_1600

File: examples/shared_samples/fc_network.rb (hunk 5 above)
Comment (AsisBagga) [inline examples/shared_samples/fc_network.rb:50]: Need to correct the class inheritance for all the api version with Synergy variant mention in this file from line 36 till 50.

File: examples/shared_samples/ethernet_network.rb (hunk 1 above)
Comment (yuvirani) [reply examples/shared_samples/ethernet_network.rb]: Updated to Synergy

File: examples/shared_samples/fc_network.rb (hunk 2 above)
Comment (yuvirani) [reply examples/shared_samples/fc_network.rb]: Fixed all together

File: spec/unit/resource/api1600/synergy/uplink_set_spec.rb (hunk 8 above)
Comment (yuvirani) [reply spec/unit/resource/api1600/synergy/uplink_set_spec.rb]: Updated to 1600

File: examples/shared_samples/network_set.rb (hunk 6 above)
Comment (yuvirani) [reply examples/shared_samples/network_set.rb]: Changed to API1600

File: examples/shared_samples/fc_network.rb (hunk 5 above)
Comment (yuvirani) [reply examples/shared_samples/fc_network.rb]: Updated as Synergy variant

**Pull request merged**